src/main/java/commands/PlanCommand.java

```diff
@@ -1,11 +1,145 @@
+        return this.fileManager;
+    }
+
+    /**
+     * Gets the instance of the PlanList class.
+     *
+     * @return An instance of the PlanList class.
+     */
+    public PlanList getPlanList() {
+        return this.planList;
+    }
+
+    /**
+     * Gets the action of the workout command specified by the user.
+     *
+     * @return A string containing the action specified by the user.
+     */
+    public String getUserAction() {
+        return this.userAction;
+    }
+
+    /**
+     * Checks for the validity of the action specified by the user and sets the class field
+     * userAction to the action (if valid).
+     *
+     * @param userAction A string containing the action specified by the user.
+     * @throws InvalidCommandException If the action specified by the user is invalid.
+     */
+    public void setUserAction(String userAction) throws InvalidCommandException {
+        switch (userAction) {
+        case CREATE_ACTION_KEYWORD:
+        case LIST_ACTION_KEYWORD:
+        case DELETE_ACTION_KEYWORD:
+        case DETAILS_ACTION_KEYWORD:
+            this.userAction = userAction;
+            break;
+        default:
+            String className = this.getClass().getSimpleName();
+            throw new InvalidCommandException(className, InvalidCommandException.INVALID_ACTION_ERROR_MSG);
+        }
+    }
+
+    /**
+     * Gets the user argument stored in the class instance.
+     *
+     * @return A string containing the user argument.
+     */
+    public String getUserArguments() {
+        return this.userArguments;
+    }
+
+
+    @Override
     public void execute() {
+        try {
+            switch (getUserAction()) {
+            case CREATE_ACTION_KEYWORD:
+                Plan newPlan = getPlanList().createAndAddPlan(getUserArguments());
+                getUI().printNewPlanCreatedMessage(newPlan);
+                break;
+            case LIST_ACTION_KEYWORD:
+                break;
+            case DETAILS_ACTION_KEYWORD:
+                break;
+            case DELETE_ACTION_KEYWORD:
+                break;
+            default:
+                String className = this.getClass().getSimpleName();
+                logger.log(Level.WARNING, "Invalid action under plan command is entered!");
+                throw new InvalidCommandException(className, InvalidCommandException.INVALID_ACTION_ERROR_MSG);
+            }
+        } catch (InvalidCommandException e) {
+            System.out.println(e.getMessage());
+            System.out.println("Please try again");
+        } catch (InvalidPlanException e) {
+            System.out.println(e.getMessage());
+            System.out.println("Please try again");
+        } catch (ArrayIndexOutOfBoundsException e) {
+            System.out.println("Uh oh, it seems like too few arguments were entered.");
+            System.out.println("Please try again. Alternatively, type 'help' if you need\n"
+                    + "more information on the commands.");
+
```

Comment:
Perhaps this empty line can be removed to be consistent with the other `catch` blocks?